src/sentry/static/sentry/app/views/releasesV2/detail/index.tsx

```diff
@@ -108,16 +113,61 @@ class ReleasesV2Detail extends AsyncView<Props, State> {
   }
 }
 
-const ReleasesV2DetailContainer = (props: Props) => (
-  <React.Fragment>
-    <GlobalSelectionHeader organization={props.organization} />
-    <ReleasesV2Detail {...props} />
-  </React.Fragment>
-);
+const ReleasesV2DetailContainer = (props: Props) => {
+  const {organization, location, router, params} = props;
+
+  const projectId = location.query.project;
+
+  // if there is no project in url, present a project selector
+  if (!projectId || typeof projectId !== 'string') {
+    openModal(
+      ({Header, Body}) => (
+        <ContextPickerModal
+          Header={Header}
+          Body={Body}
+          needOrg={false}
+          needProject
+          nextPath={`/organizations/${organization.slug}/releases-v2/${encodeURIComponent(
+            params.release
+          )}/?project=:project`}
+          onFinish={pathname => {
+            router.replace(pathname);
+          }}
+        />
+      ),
+      {
+        onClose() {
+          // if a user closes the modal (either via button, Ecs, clicking outside)
+          router.push(`/organizations/${organization.slug}/releases-v2/`);
+        },
+      }
+    );
+
+    return <ContextPickerBackground />;
+  }
+
+  // otherwhise business as usual
+  return (
+    <React.Fragment>
+      <GlobalSelectionHeader
+        organization={organization}
+        shouldForceProject
+        lockedMessageSubject={t('release')}
+        forceProject={organization.projects.find(p => p.id === projectId)}
```

Comment:
Can we refactor this to use `app/utils/projects` - we are in the process of removing `projects` from the `organization` details because of performance reasons.